File: GrooverAdmSPA/ClientApp/src/app/places/models/Playlist.ts

export class Playlist {
  public name?: string;
  public url?: string;
  public imageUrl?: string;
  public checked?: boolean;

  toJson(data?: any) {
    data = typeof data === 'object' ? data : {};
    data['name'] = this.name;
    data['url'] = this.url;
    data['imageURl'] = this.imageUrl;
    data['checked'] = this.checked;
    return data;
  }
}

File: GrooverAdm.DataAccess.Firestore.Model/Model/Playlist.cs
using Google.Cloud.Firestore;
using System;
using System.Collections.Generic;
using System.Text;

namespace GrooverAdm.DataAccess.Firestore.Model
{
    [FirestoreData]
    public class Playlist : DataAccess.Model.Playlist
    {

        [FirestoreDocumentId]
        public DocumentReference Reference { get; set; }
        [FirestoreProperty]
        public string Id { get; set; }
        [FirestoreProperty]
        public string Name { get; set; }
        [FirestoreProperty]
        public string ImageUrl { get; set; }
        [FirestoreProperty]
        public string SnapshotId { get; set; }
        [FirestoreProperty]
        public string Url { get; set; }
        [FirestoreProperty]
        public Dictionary<string, int> LastFmOcurrenceDictionary { get; set; }
        [FirestoreProperty]
        public Dictionary<string, int> SpotifyOccurrenceDictionary { get; set; }
    }
}
